### PR TITLE
v2.1: simplified Dagger management and module architecture

### PR DESCRIPTION
**Pull request by marcosalis**
Pretty significant overhaul in the way the Dagger dependency graph is managed:
- Each layer/module manages and initialises its own component
- No more `business-injection` and "bridge" modules
- Each layer exposes a component interface with no direct reference to Dagger
- Added a `data-bridge` module with a single class with the only purpose of initialising the data layer

Updated README:
- New modules and components diagrams
- Simplified explanation of how Dagger works in the sample

Other changes:
- Lots of classes converted to Kotlin
- Dagger updated to latest version (`2.25.2`)
- Added Bitrise C.I. builds support and status badge in README

### CODE REVIEW

File: sample-business/src/main/java/com/teamwork/android/samples/clean/business/SampleBusinessApplication.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     */
+    protected abstract fun initializeCacheLayer()
+
+    protected fun initializeBusinessLayer() {
+        businessLayerInitializer.initializeBusinessLayer(this)
+    }
+
+    @CallSuper
+    protected open fun onDependencyManagementInitialized() {
+        initializeLayers()
```
Comment (aidangrabe) [inline sample-business/src/main/java/com/teamwork/android/samples/clean/business/SampleBusinessApplication.kt]: `initializeLayers()` gets called already in `onCreate`, doesn't need to be called again here does it?

File: sample-business/src/main/java/com/teamwork/android/samples/clean/business/feature1/detail/Feature1DetailsInteractorImpl.kt [hunk 2]
```diff
@@ -2,12 +2,15 @@
...
 package com.teamwork.android.samples.clean.business.feature1.detail
 
+import android.content.Context
 import com.teamwork.android.samples.clean.business.BaseInteractor
 import com.teamwork.android.samples.clean.business.Interactor
 import com.teamwork.android.samples.clean.data.access.feature1.Entity1Repo
 import com.teamwork.android.samples.clean.entity.feature1.Entity1
 import javax.inject.Inject
 
 class Feature1DetailsInteractorImpl
```
Comment (aidangrabe) [inline sample-business/src/main/java/com/teamwork/android/samples/clean/business/feature1/detail/Feature1DetailsInteractorImpl.kt]: Could make this guy `internal` and same for `Feature1DetailsInteractorImpl`?

File: sample-business/src/main/java/com/teamwork/android/samples/clean/business/SampleBusinessApplication.kt (hunk 1 above)
Comment (marcosalis) [reply sample-business/src/main/java/com/teamwork/android/samples/clean/business/SampleBusinessApplication.kt]: Good spot! 😱

**Review by aidangrabe — APPROVED** [review]
🌰

**Pull request merged**